Sources/PushNotifications.swift

```diff
@@ -8,26 +8,45 @@ import NotificationCenter
 import Foundation
 
 @objc public final class PushNotifications: NSObject {
-    private var deviceId: String?
-    private var instanceId: String?
-    private var baseURL: String?
     private let session = URLSession.shared
     //! Returns a shared singleton PushNotifications object.
     /// - Tag: shared
     @objc public static let shared = PushNotifications()
 
     /**
-     Register with PushNotifications service.
+     Start PushNotifications service.
 
      - Parameter instanceId: PushNotifications instance id.
 
      - Precondition: `instanceId` should not be nil.
      */
-    @objc public func register(instanceId: String) {
-        self.instanceId = instanceId
-        self.baseURL = "https://\(instanceId).pushnotifications.pusher.com/device_api/v1/instances"
-        self.registerForPushNotifications()
+    @objc public func start(instanceId: String) {
+        do {
+            try Instance.persist(instanceId)
+        }
+        catch PusherAlreadyRegisteredError.instanceId(let errorMessage) {
```

Comment:
Nitpick: Normally `do catch` blocks are done like this:

```swift
do {
    try Instance.persist(instanceId)
} catch PusherAlreadyRegisteredError.instanceId(let errorMessage) {
    print(errorMessage)
} catch { 
    print("Unexpected error: \(error).")
}
```

Also, and I'm not sure about this one, do you want to interpolate the `error` or do you want to call `error.localizedDescription`?